packages/app/src/explorer/utils.ts

```diff
@@ -0,0 +1,24 @@
+import type { Entity } from '@h5web/shared';
+import { assertStr, isGroup } from '@h5web/shared';
+
+import { getAttributeValue, hasAttribute } from '../utils';
+
+const SUPPORTED_NX_CLASSES = new Set(['NXdata', 'NXentry', 'NXprocess']);
+
+export function isNxGroup(entity: Entity): boolean {
+  if (!isGroup(entity)) {
+    return false;
+  }
+
+  if (hasAttribute(entity, 'default')) {
+    return true;
+  }
+
+  if (hasAttribute(entity, 'NX_class')) {
+    const nxClass = getAttributeValue(entity, 'NX_class');
+    assertStr(nxClass);
+    return SUPPORTED_NX_CLASSES.has(nxClass);
+  }
+
+  return false;
```

Comment:
Obviously, this algorithm is not as accurate as the default entity resolution algorithm: if an entity has an implicit default child but doesn't have an `NX_class`, it won't have the NX badge (whereas it should). Conversely, if an `NXprocess` entity doesn't have an implicit default child, it will have the NX badge (whereas it shouldn't). But it will do for now.

Reply:
That's correct. What I find annoying is that we get both false positives (badge shown when it shouldn't) and false negatives (badge not shown when it should). We'll have to see if someone complains about either when confronted to real files.

Reply:
These cases should be documented in an issue so that we can see how to move forward